### PR TITLE
feat(updaters): add Maven pom.xml updater with property indirection

### PR DESCRIPTION
**Pull request by ruromero**
## Summary
- New `src/updaters/maven_updater.js` module with `updateMavenVersions()` API
- Position-based string splice preserves all formatting, comments, whitespace
- Supports `${property}` indirection with recursive resolution and cycle detection
- Handles `<dependencyManagement>` sections
- 17 new tests, 6 fixture pom.xml files, 89% line coverage

Implements [TC-5411](https://redhat.atlassian.net/browse/TC-5411)

## Test plan
- [x] Direct version replacement
- [x] Property-based version replacement
- [x] Recursive property chain resolution
- [x] Circular property detection and skip
- [x] Formatting/comment preservation
- [x] Malformed XML fail-safe
- [x] Idempotency verification
- [x] `npm run lint` passes
- [x] `npm test` passes

[TC-5411]: https://redhat.atlassian.net/browse/TC-5411?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by Sourcery

Add a Maven pom.xml updater that performs structure-aware, formatting-preserving dependency version updates with robust handling of property indirection and edge cases, backed by comprehensive tests and fixtures.

New Features:
- Introduce a Maven pom.xml updater that updates dependency versions by groupId/artifactId while preserving file formatting and comments
- Support version updates via Maven property references, including recursive property chains and dependencyManagement entries

Enhancements:
- Detect and skip circular or conflicting property-based version updates, reporting detailed reasons for skipped changes
- Ensure idempotent behavior and graceful handling of malformed or non-standard pom.xml structures

Tests:
- Add a comprehensive maven_updater test suite with fixtures covering direct versions, property indirection, recursive and circular properties, formatting preservation, error handling, idempotency, and dependencyManagement support

### CODE REVIEW

**Review by a-oren — APPROVED** [review]
LGTM

**Comment by ruromero** [comment]
[sdlc-workflow/verify-pr] Re: @sourcery-ai[bot] review — 

1. "duplicate direct dep conflict check" — Classified as **suggestion** — already addressed by commit dcfb3d3 which adds dedup checking to the direct-version path, aligning it with the property path. No sub-task created.

2. "relaxing string-based tag searches" — Classified as **suggestion** — this proposes a future robustness improvement for namespace-prefixed or attribute-bearing XML tags. No documented CONVENTIONS.md pattern or demonstrated codebase convention matches this suggestion. No sub-task created.

File: test/updaters/maven_updater.test.js [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+
+			// Then
+			expect(result.applied).to.have.lengthOf(0);
+			expect(result.skipped).to.have.lengthOf(1);
+			expect(result.skipped[0].reason).to.include('No <project> element');
+			expect(result.content).to.equal(pom);
+		});
+	});
+
+	suite('idempotency', () => {
```
[the remark this answers is not in the document]
Comment (ruromero) [reply test/updaters/maven_updater.test.js:331]: [sdlc-workflow/verify-pr] Classified as **suggestion** — this proposes expanding test assertions for idempotency metadata (applied/skipped arrays). No documented CONVENTIONS.md pattern prescribes assertion depth for idempotency tests. No sub-task created.

**Comment by ruromero** [comment]
## Verification Report for TC-5411 (commit dcfb3d3)

| Check | Result | Details |
|-------|--------|---------|
| Review Feedback | PASS | 3 suggestions from sourcery-ai[bot]; no code change requests |
| Root-Cause Investigation | N/A | No sub-tasks created |
| Scope Containment | PASS | All 8 PR files within task-specified scope |
| Diff Size | PASS | 911 additions across 8 new files; proportionate (39.7% impl, 44.6% tests, 15.7% fixtures) |
| Commit Traceability | WARN | Commit 1 (fcda80e) references TC-5411; commit 2 (dcfb3d3) does not |
| Sensitive Patterns | PASS | No secrets detected in 768 added lines across 8 files |
| CI Status | PASS | All 5 CI checks pass (lint, test Node 22/24, PR title, commit messages, Sourcery) |
| Acceptance Criteria | PASS | 8 of 8 criteria met |
| Test Quality | PASS | No repetitive tests; all 17 tests have doc comments; Eval Quality: N/A |
| Test Change Classification | ADDITIVE | New test file + 6 fixture files; no existing tests modified |
| Verification Commands | PASS | `npm run lint` and `npm test` pass via CI |

### Overall: WARN

Commit `dcfb3d3` ("fix(updaters): detect conflicting duplicate direct version changes") does not reference TC-5411 in its message. This is a follow-up fix addressing sourcery-ai's review suggestion about duplicate direct dep conflict detection. All other checks pass.

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/RHEcosystemAppEng/sdlc-plugins) v0.13.7.*
